fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/serialization/LoanValidatorImpl.java

```diff
@@ -0,0 +1,79 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.portfolio.loanaccount.serialization;
+
+import com.google.gson.JsonElement;
+import java.math.BigDecimal;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.fineract.infrastructure.core.api.JsonCommand;
+import org.apache.fineract.infrastructure.core.exception.InvalidJsonException;
+import org.apache.fineract.infrastructure.core.serialization.FromJsonHelper;
+import org.apache.fineract.infrastructure.core.service.MathUtil;
+import org.apache.fineract.portfolio.common.service.Validator;
+import org.apache.fineract.portfolio.loanaccount.api.LoanApiConstants;
+import org.apache.fineract.portfolio.loanaccount.domain.Loan;
+import org.apache.fineract.portfolio.loanaccount.domain.LoanDisbursementDetails;
+import org.apache.fineract.portfolio.loanaccount.domain.LoanRepository;
+import org.apache.fineract.portfolio.loanaccount.exception.LoanNotFoundException;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public final class LoanValidatorImpl implements LoanValidator {
+
+    private final FromJsonHelper fromApiJsonHelper;
+    private final LoanRepository loanRepository;
+
+    @Override
+    public void validateLoanApprovedAmountModification(JsonCommand command) {
+        String json = command.json();
+        if (StringUtils.isBlank(json)) {
+            throw new InvalidJsonException();
+        }
+
+        Validator.validateOrThrowDomainViolation("loan.approved.amount", baseDataValidator -> {
+            final JsonElement element = this.fromApiJsonHelper.parse(json);
+            final BigDecimal newApprovedAmount = this.fromApiJsonHelper
+                    .extractBigDecimalWithLocaleNamed(LoanApiConstants.amountParameterName, element);
+
+            baseDataValidator.reset().parameter(LoanApiConstants.amountParameterName).value(newApprovedAmount).positiveAmount();
+
+            final Long loanId = command.getLoanId();
+            Loan loan = this.loanRepository.findById(loanId).orElseThrow(() -> new LoanNotFoundException(loanId));
+            BigDecimal approvedPrincipal = loan.getApprovedPrincipal();
+
+            if (MathUtil.isGreaterThan(newApprovedAmount, approvedPrincipal)) {
```

Comment:
We should allow to increase approved amount if:
- Applied amount + configured threshold  equals or higher than "new approved" amount
- new approved amount cannot be lower than already disbursed amount + total capitalized income
- new approved amount cannot be lower than total "expected disbursement" + total capitalized income